File: 0011-container-with-most-water/0011-container-with-most-water.js
/**
 * @param {number[]} height
 * @return {number}
 */
var maxArea = function(height) {
    let ans = 0;
    let x=0;
    let y=height.length-1;
    while ( x<y ){
        let l = Math.min(height[x],height[y])
        let b = y-x;
        ans = Math.max(l*b,ans)
        
        if(height[x]<height[y]){
            x++;
        }else{
            y--;
        }
    }
    
    return ans;
};

File: 0011-container-with-most-water/0011-container-with-most-water.py
class Solution:
    def maxArea(self, height: List[int]) -> int:

        x = 0
        y = len(height)-1
        ans = 0

        while (x<y):
            l = min(height[x],height[y])
            b = y - x
            ans =max( ans, l*b )

            if height[x] > height[y]:
                y-=1
            else:
                x+=1
        return ans